build.gradle

```diff
@@ -468,6 +468,79 @@ project(':iceberg-mr') {
   }
 }
 
+if (jdkVersion == '8') {
```

Comment:
Why only in Java 8?

Reply:
Unfortunately Hive3 only works with Java8. There are a number of classloader related issues when running Hive3 on Java11.

Reply:
So HIve 2 works with newer versions of Java than Hive 3 does? :facepalm:

Reply:
Yeah, what can I say... 😢  The upgrade work is ongoing and I believe is tracked here: https://issues.apache.org/jira/browse/HIVE-22415

Reply:
Do tests actually fail with JDK 11? Or is this because they might fail?

Reply:
Yes, all the `TestHiveIcebergStorageHandlerWith*` tests in `iceberg-hive3` fail on JDK11 with:
```
java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader
```
This issue is fixed in https://issues.apache.org/jira/browse/HIVE-21584, but it's not part of the official Hive 3.1.2 release, it's only in master as of now.